core/io/resource_loader.cpp

```diff
@@ -102,8 +102,11 @@ void ResourceFormatLoader::get_recognized_extensions_for_type(const String &p_ty
 }
 
 void ResourceLoader::get_recognized_extensions_for_type(const String &p_type, List<String> *p_extensions) {
+	StringName native = ScriptServer::is_global_class(p_type) ? ScriptServer::get_global_class_native_base(p_type) : StringName(p_type);
 	for (int i = 0; i < loader_count; i++) {
-		loader[i]->get_recognized_extensions_for_type(p_type, p_extensions);
+		Ref<ResourceFormatLoader> current = loader[i];
+		String type = !current->get_script().is_null() ? p_type : String(native);
+		current->get_recognized_extensions_for_type(type, p_extensions);
 	}
```

Comment:
Yes, this is true. I guess the intention here was to check whether the resource for the given type was a custom resource (so it would have a script attached). If yes, it would return the custom resource. If not, the resource it extends...

@willnationsdev Any ideas about this?

Reply:
Hmm, trying to recall exactly what I was doing. I think I was contending with problems regarding how the core ResourceFormatLoaders would make assumptions about handling resources, even custom ones, that were resulting in exceptions being thrown. And I needed to get them to treat my resource (extending an in-engine type) the way they would have evaluated any other in-engine type with the same base class. So I simply said, "if you're a ResourceFormatLoader with a script, then treat it however you want. But if you don't have a script, then just think of it like you would its base class." And whatever I was testing at the time worked with that setup.

> in reality, passing it anything other than a .gluten causes the computer to violently explode.

I don't think this is necessarily a major concern (though I could be wrong). In the same way that you can give anything a .res extension and it's expected to be able to load successfully, even if it doesn't have the features you want. Furthermore, if you are *extending* a class that expects a given extension, then I would expect that class to have some level of compatibility with the base extensions (like `.res` with all `Resource` types). Otherwise, you should've extended a completely different type.

Reply:
So let's keep it in unless a core dev disagrees.